code/modules/mob/living/simple_animal/bot/floorbot.dm

```diff
@@ -90,6 +91,28 @@
 
 	return dat
 
+/mob/living/simple_animal/bot/floorbot/attackby(obj/item/W , mob/user, params)
+	if(istype(W, /obj/item/stack/tile/plasteel))
+		user << "<span class='notice'>The floorbot can produce normal tiles itself.</span>"
+		return
+	if(specialtiles != 0 && istype(W, /obj/item/stack/tile/))
+		var/obj/item/stack/tile/usedtile = W
+		if(usedtile != tiletype)
+			user << "<span class='warning'>Different custom tiles are already inside the floorbot.</span>"
+			return
+	if(istype(W, /obj/item/stack/tile/))
+		tiletype = W
```

Comment:
very much don't like this, you're making a reference to the tile, but then leaving it in the user's hands and absolutely ANYTHING could happen to that tile (most resulting in it being deleted) which would in turn cause the bot to lose reference to its custom tiles.

I would recommend having ```tiletype``` actually BE a typepath, not a reference to an instance of a type.